### ISSUE TITLE
Missing the actual magic

### ISSUE THREAD

**Issue by stappersg** [ap51-flash/ap51-flash#10]
Hi,

Neither the `README.md` nor  _readthedocs_ say any thing about the actual working of the program.

There is a vague hint on _redboot_  and _you don't need TFTP server_.

But what is  **ap51-flash**?

Please document what goes over the wire.  With _wire_ I do mean ethernet cable.

If _redboot_ already has document this, then please do deep link to the redboot information.

Thank you for considering this.

**Comment by ecsv** [comment]
Yes, we definitely have to have add useful information to the documentation. This was neglected for years. And I have only added the current sphinx documentation system  as starting point when I've assisted Marek during the move of ap51-flash to here. It was previously in a quite closed and hidden development environment and Marek wasn't anymore allowed to work on this system. So I just scraped everything I could find & was allowed to use and put it in this minimal version.

I cannot prioritize this task at the moment in my queue. But interested persons might want to pick up. So here are some pointer of things which would be interesting in the flashing process:

* device detection (ARP packet sniffing)
* device identification ((gratuitous) ARP based hints, ... - just check detect_main functions)
* start of flashing process (just wait for TFTP request to send instruction file, active netconsole connection, redboot telnet connection, active tftp push, ...)
* transfer of image parts (send data like a TFTP client to the device, answer TFTP requests like a TFTP server)


Btw. (patched) redboot is only one supported remote bootloaders. On newer systems, you will usually see an (modified) u-boot. There are various ones which either have a complete automated system (like the Plasmacloud devices) which just requires an intelligent (to select correct image + parse them) TFTP server, some which just need to get the image uploaded using TFTP client and some which actually need command injection via netconsole or similar systems.
